artemis/modules/SubdomainEnumeration.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+import time
+from typing import Set, Callable
+
+from karton.core import Task
+
+from artemis.binds import TaskStatus, TaskType
+from artemis.config import Config
+from artemis.domains import is_subdomain
+from artemis.module_base import ArtemisBase
+from artemis.utils import check_output_log_on_error
+
+DOMAIN_REGEX = r"([a-z0-9\-]+\.)+[a-z0-9\-]+"
+
+# NOTE: The rappidns, jldc, gau and crtsh modules were removed from this class 
+# as their functionality is already implemented internally by the subfinder and amass utilities.
+
+
+class SubdomainEnumeration(ArtemisBase):
+    """
+    Consumes `type: domain` to gather subdomains and produces `type: domain`.
+    """
+
+    identity = "SubdomainEnumeration"
+    filters = [
+        {"type": TaskType.DOMAIN.value},
+    ]
+    lock_target = False
+
+    def get_subdomains_with_retry(self, func: Callable[[str], Set[str]], domain: str, retries: int = 20, sleep_time: int = 300) -> Set[str]:
+        for retry in range(retries):
+            try:
+                return func(domain)
+            except Exception:
```

Comment:
hmm, this can't possibly work. you catch exceptions but each of get_subdomains_from won't raise an exceptino as its internals are wrapped inside try...ecxcept